dot/sync/syncer.go

```diff
@@ -434,6 +448,28 @@ func (s *Service) handleRuntimeChanges(newState *rtstorage.TrieState) error {
 	return nil
 }
 
+func (s *Service) handleCodeSubstitution(block common.Hash) error {
+	for key, value := range s.codeSubstitute {
```

Comment:
check if this is in map instead of iterating

Reply:
Updated.